crates/sp-lightclient/src/lib.rs

```diff
@@ -304,18 +304,23 @@ impl<Header: HeaderT, Store: Storage<Header>> HeaderImporter<Header, Store> {
         SubspaceDigestItems<FarmerPublicKey, FarmerPublicKey, FarmerSignature>,
         ImportError<Header>,
     > {
+        // extract digest items from the header
         let pre_digest_items = extract_subspace_digest_items(header)?;
-        if pre_digest_items.global_randomness != parent_header.derived_global_randomness {
+        // extract next digest items from the parent header
+        let next_digest_items = parent_header.extract_next_digest_items()?;
```

Comment:
This will fail for genesis block though. I don't have the whole code in my head right now, is that expected?
If not then this needs to be handled explicitly by using derived parameters of the parent block for genesis instead.